### PR TITLE
Use OAuth for the extension version too

### PR DESCRIPTION
**Pull request by bhollis**
This switches from using the bungled token to always using OAuth, even for the extension. With the token removed, I was able to do a lot of cleanup to the `dimBungieService` too!

### CODE REVIEW

File: src/scripts/services/dimBungieService.factory.js [hunk 1]
```diff
@@ -292,42 +212,18 @@ function BungieService($rootScope, $q, $timeout, $http, $state, dimState, toaste
...
+        )))
+        .then(handleErrors, handleErrors)
+        .then(processBnetCharactersRequest);
 
     return charactersPromise;
 
-    function getBnetCharactersRequest(token, platform, membershipId) {
-      return {
-        method: 'GET',
-        url: 'https://www.bungie.net/Platform/Destiny/Tiger' + (platform.type === 1 ? 'Xbox' : 'PSN') + '/Account/' + membershipId + '/',
```
Comment (SunburnedGoose) [inline src/scripts/services/dimBungieService.factory.js]: You can remove the 'Tiger' trinary here and just use the platform.type (1 or 2) for Xbox and PSN.

https://destinydevs.github.io/BungieNetPlatform/docs/API-Test#/GetAccount/p:membershipType=1

File: src/scripts/services/dimBungieService.factory.js [hunk 2]
```diff
@@ -292,42 +212,18 @@ function BungieService($rootScope, $q, $timeout, $http, $state, dimState, toaste
...
-      .then(getMembershipPB)
-      .then(function(membershipId) {
-        return getBnetCharactersRequest(data.token, platform, membershipId);
-      })
-      .then($http)
-      .then(handleErrors, handleErrors)
-      .then(processBnetCharactersRequest);
+    var charactersPromise = getMembership(platform)
+        .then((membershipId) => $http(bungieApiQuery(
+          `/Platform/Destiny/Tiger${platformName}/Account/${membershipId}/`
```
Comment (SunburnedGoose) [inline src/scripts/services/dimBungieService.factory.js]: You can remove the 'Tiger' trinary here and just use the platform.type (1 or 2) for Xbox and PSN.

https://destinydevs.github.io/BungieNetPlatform/docs/API-Test#/GetAccount/p:membershipType=1

**Comment by SunburnedGoose** [comment]
Was this the PR that uses 

https://www.bungie.net/Platform/User/GetCurrentBungieAccount/

to get all the Destiny accounts, characters, & platforms?

File: src/scripts/services/dimBungieService.factory.js [hunk 3]
```diff
@@ -346,72 +242,29 @@ function BungieService($rootScope, $q, $timeout, $http, $state, dimState, toaste
...
-      .then(addTokenToData)
-      .then(getMembershipPB)
-      .then(addMembershipIdToData)
-      .then(getCharactersPB)
-      .then(addCharactersToData)
-      .then(function() {
-        var promises = [
-          getDestinyInventories(data.token, platform, data.membershipId, data.characters),
-          getDestinyProgression(data.token, platform, data.membershipId, data.characters)
+    return $q.all([
```
Comment (kyleshay) [inline src/scripts/services/dimBungieService.factory.js]: nice... firing both async. should help slightly with load?

nice work on refactoring these functions btw!

**Comment by kyleshay** [comment]
nice! unified flow makes the most sense going forward.

few other places where `window.chrome.extension` exist in the codebase. could consider just removing them all as part of this PR.

could also remove references to `$DIM_API_KEY`

**Comment by bhollis** [comment]
Thanks - all the stuff you guys are suggesting is in a followup PR that depends on this one.

**Pull request merged**